### DISCUSSION TITLE
Big steps in training data makes model fail to learn

### DISCUSSION THREAD

**Discussion by gabus** [mrdbourke/pytorch-deep-learning#728] [Q&A]
General question. This is linear regression model.
I would expect optimiser function to only care about model's parameters. Why train steps has influence in learning?

Model parameters:
```
bias = 0.3
weight = 0.7
```

Training steps which work ok:
```
start = 0
end = 1
step = 0.01

X = torch.arange(start, end, step).unsqueeze(dim=1) 
y = X * weight + bias
```

Training steps which makes the model get stuck:
```
start = 0
end = 100
step = 10.0
```

**Comment by shisirkha** [comment]
example:
a kid saw a coin. he is giving larger and larger step to get the coin, but he has 1 disability that he cannot move his hand back so he will pass the coin and never reach it.

this example is same as taking larger steps in machine learning. if you take larger and larger steps the model will pass the results, but it will never reach that actual result.  so recommended learning rate is 0.01 or 0.001
